kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
```

Comment:
@thibaultcha 

here's one: http://stackoverflow.com/questions/30967822/when-do-i-use-path-params-vs-query-params-in-a-restful-api

the 404 related comment in the top answer here seems legit: http://stackoverflow.com/questions/4024271/rest-api-best-practices-where-to-put-parameters

That said from the overall kong api now, the path based solution seems easier to use, though it might be not as restful. So effectively I do not have a preference.